arangod/Agency/NodeLoadInspector.h

```diff
@@ -0,0 +1,311 @@
+      return {e.what()};
+    }
+  }
+
+  [[nodiscard]] Status value(std::string& v) {
+    if (auto value = _node->getString(); value) {
+      v = std::move(*value);
+      return {};
+    }
+    return {"Expecting type String"};
+  }
+
+  [[nodiscard]] Status value(std::string_view& v) {
+    static_assert(AllowUnsafeTypes);
+    if (!_node->isString()) {
+      return {"Expecting type String"};
+    }
+    v = *_node->getStringView();
+    return {};
+  }
+
+  [[nodiscard]] Status value(velocypack::HashedStringRef& v) {
+    static_assert(AllowUnsafeTypes);
+    if (!_node->isString()) {
+      return {"Expecting type String"};
+    }
+    auto s = *_node->getStringView();
+    if (s.size() > std::numeric_limits<uint32_t>::max()) {
+      return {"String value too long to store In HashedStringRef"};
+    }
+    v = velocypack::HashedStringRef(s.data(), static_cast<uint32_t>(s.size()));
+    return {};
+  }
+
+  [[nodiscard]] Status::Success value(velocypack::Slice& v) {
+    static_assert(AllowUnsafeTypes);
+    v = _node->slice();
+    return {};
+  }
+
+  [[nodiscard]] Status value(bool& v) {
+    if (auto value = _node->getBool(); value) {
+      v = std::move(*value);
+      return {};
+    }
+    return {"Expecting type Bool"};
+  }
+
+  [[nodiscard]] Status beginObject() {
+    if (!_node->isObject()) {
+      return {"Expecting type Object"};
+    }
+    return {};
+  }
+
+  [[nodiscard]] Status::Success endObject() { return {}; }
+
+  [[nodiscard]] Status beginArray() {
+    if (!_node->isArray()) {
+      return {"Expecting type Array"};
+    }
+    return {};
+  }
+
+  [[nodiscard]] Status::Success endArray() { return {}; }
+
+  bool isNull() const noexcept {
+    return _node == nullptr || _node->slice().isNull();
+  }
+
+  consensus::Node const* node() const noexcept { return _node; }
+
+ private:
+  template<class>
+  friend struct detail::EmbeddedFields;
+  template<class, class...>
+  friend struct detail::EmbeddedFieldsImpl;
+  template<class, class, class>
+  friend struct detail::EmbeddedFieldsWithObjectInvariant;
+  template<class, class>
+  friend struct detail::EmbeddedFieldInspector;
+
+  template<class, class, class>
+  friend struct LoadInspectorBase;
+
+  auto make(velocypack::Slice data) {
+    if constexpr (Base::hasContext) {
+      return VPackLoadInspectorImpl<AllowUnsafeTypes, Context>(
+          data, this->options(), this->getContext());
+    } else {
+      return VPackLoadInspectorImpl<AllowUnsafeTypes>(data, this->options());
+    }
+  }
+
+  auto getTypeTag() const noexcept {
+    assert(_node->type() == consensus::LEAF);
+    return _node->slice().type();
+  }
+
+  template<class Func>
+  auto doProcessObject(Func&& func)
+      -> decltype(func(std::string_view(),
+                       std::declval<consensus::Node const*>())) {
+    assert(_node);
+    assert(_node->isObject());
+    for (auto& [k, v] : _node->children()) {
+      if (auto res = func(k, v.get()); not res.ok()) {
```

Comment:
I think we have codestyle, why do you write code in other style?

Is it too difficult to write code in same style in same directory? :(

Reply:
I am not sure if we actually have this in our coding guidelines. Either way, we have quite some code that uses word operators, and I find it much more readable. In particular a single `!` can easily be overlooked. You are right, that the style in this file/directory is not fully consistent, because some of it is older code. I can happily update everything to use word operators though.

Reply:
> and I find it much more readable

For me any is ok, but in general seeing not/and/or in c++ code is very confusing, I always knew it works, but almost no one write like this. Personally I know only replication 2 team which start to write like this in last year

Reply:
Yes, I also used to write "classic" C++ code, but in particular the logical not operator `!` often caused confusion because I misread some condition. A lot of other languages (Ruby, Python, ...) use word operators which I find much more readable. And since we also have those in C++, I would like to use them. But I agree that we should try to establish a common style.